cirq-core/cirq/ops/pauli_string.py

```diff
@@ -501,7 +501,9 @@ def matrix(self, qubits: Optional[Iterable[TKey]] = None) -> np.ndarray:
         """
         qubits = self.qubits if qubits is None else qubits
         factors = [self.get(q, default=identity.I) for q in qubits]
-        return linalg.kron(self.coefficient, *[protocols.unitary(f) for f in factors])
```

Comment:
for parameters like this, I think we should check to see if the object is parmaeterized (`cirq.is_parameterized(self)`) and raise an exception if it is.  In other words this is actually sort of catching a common bug (we forget to check if the object is parameterized).  If not, then we should assert it is complex instead of a cast.  (Same for the example below as well)

Reply:
Done - added check for parameterization to `PauliString.matrix()` and to `GlobalPhaseGate._apply_unitary_()`.
I left the casting to np.generic in global_phase_op.py and phased_iswap_gate.py as it was, because the mypy complaint there stems from a possibility of numbers.Complex type, which is not included in numpy functions input.
(At those codepoints the argument is assured to be a number and cannot be sympy.Expr)